examples/vision/siamese.py

```diff
@@ -0,0 +1,345 @@
+          labels[i],
+          predictions[i][0]))
+        else:
+          ax.set_title('Label: {}'.format(labels[i]))
+  if test:
+    plt.tight_layout(rect = (0,0,1.9,1.9 ), w_pad=0.0)
+  else:
+    plt.tight_layout(rect = (0,0,1.5,1.5))
+  plt.show()
+
+
+
+
+
+visualize(train_ds, to_show=3)
+
+"""
+## Define the model
+
+There will be two input layers, each leading to its own network, which
+produces embeddings. Lambda layer will merge them using
+[Euclidean distance](https://en.wikipedia.org/wiki/Euclidean_distance) and the
+merged layer will be fed to final network.
+"""
+
+
+# Provided two tensors t1 and t2
+# Euclidean distance = sqrt(sum(square(t1-t2)))
+def euclidean_distance(vects):
+    """
+    Parameter
+    ---------
+    vect  : list
+            list containing two tensors of same length
+
+    Return
+    ------
+    Tensor containing euclidean distance
+    (as floating point value) between vectors
+    """
+
+    x, y = vects
+    sum_square = K.sum(K.square(x - y), axis=1, keepdims=True)
+    return K.sqrt(K.maximum(sum_square, K.epsilon()))
+
+
+
+input = Input((28,28,1))
+x = tf.keras.layers.BatchNormalization()(input)
+x = Conv2D(4, (5,5), activation = 'tanh')(x)
+x = AveragePooling2D(pool_size = (2,2))(x)
+x = Conv2D(16, (5,5), activation = 'tanh')(x)
+x = AveragePooling2D(pool_size = (2,2))(x)
+x = Flatten()(x)
+
+x = tf.keras.layers.BatchNormalization()(x)
+x = Dense(10, activation = 'tanh')(x)
+embedding_network = Model(input, x)
+
+
+input_1 = Input((28,28,1))
+input_2 = Input((28,28,1))
+
+# As mentioned above, Siamese Network share weights between
+# tower networks (sister networks). To allow this, we will use
+# same embedding network for both tower networks.
+tower_1 = embedding_network(input_1)
+tower_2 = embedding_network(input_2)
+
+merge_layer = Lambda(euclidean_distance)([tower_1, tower_2])
+normal_layer = tf.keras.layers.BatchNormalization()(merge_layer)
+output_layer = Dense(1, activation="sigmoid")(normal_layer)
+siamese = Model(inputs=[input_1, input_2], outputs=output_layer)
+
+# Contrastive loss = mean( (1-true_value) * square(prediction) +
+#                         true_value * square( max(margin-prediction, 0) ))
+def contrastive_loss(y_true, y_pred, margin=1):
+    """
+    Parameters
+    ----------
+    margin  : int
+              margin defines the baseline for distance for which pairs
+              should be classified as dissimilar. - (default is 1)
+
+    y_true  : list
+              list of labels, each label is of type float32
+
+    y_pred  : list
+              list of predictions of same length as of y_true,
+              each label is of type float32
+    
+    Return
+    ------
+    A tensor containing constrastive loss as floating point value
+    """
+
+    square_pred = K.square(y_pred)
+    margin_square = K.square(K.maximum(margin - (y_pred), 0))
+    return K.mean((1-y_true) * square_pred + (y_true) * margin_square)
+
+siamese.compile(
+	loss = contrastive_loss, 
```

Comment:
Make sure to use `black` to format the `.py` file. For instance, there should not be spaces around `=`.